core/utils/xml.js

```diff
@@ -24,6 +24,13 @@ goog.provide('Blockly.utils.xml');
  */
 Blockly.utils.xml.NAME_SPACE = 'https://developers.google.com/blockly/xml';
 
+/**
+ * Matches control characters tha need to be replaced with NCRs (numerical
+ * character references) before serializing to XML.
+ */
+// eslint-disable-next-line no-control-regex
+Blockly.utils.xml.INVALID_CONTROL_CHARS = /[\u0000-\u0009\u000B\u000C\u000E-\u001F]/;
```

Comment:
Suggest: [\x00-\x09\x0B\x0C\x0E-\x1F]
It's shorter, and it also reduces future temptations to add Unicode ranges above FF which may break sanitizeText which can't handle multi-byte Unicode characters.